### PR TITLE
Add a management command to update experiment metadata

### PR DESCRIPTION
**Pull request by wvauclain**
## Issue Number

#1395 

## Purpose/Implementation Notes

I added a new management command to update the metadata for experiments. I also had to rename the old one that updated sample metadata so that the names weren't confusing, and I made some other changes to that command based off of what I learned writing this one.

## Types of changes

What types of changes does your code introduce?

- New feature (non-breaking change which adds functionality)

## Functional tests

I wrote a test to make sure that experiments' titles are updated when running this command.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Screenshots

Please attach any screenshots that illustrate these changes.

### CODE REVIEW

File: foreman/data_refinery_foreman/foreman/management/commands/update_sample_metadata.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--source-database",
+            type=str,
+            help=("The name of a source database, such as ARRAY_EXPRESS, GEO, or SRA."
+                  "All samples from this source database will have their metadata refreshed."))
+
+    def handle(self, *args, **options):
+        """Refreshes the metadata for RNA-Seq samples
```
Comment (kurtwheeler) [inline foreman/data_refinery_foreman/foreman/management/commands/update_sample_metadata.py]: This says 'RNA-Seq` but there's an option to specify the source database.

File: foreman/data_refinery_foreman/foreman/management/commands/update_experiment_metadata.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--source-database",
+            type=str,
+            help=("The name of a source database, such as ARRAY_EXPRESS, GEO, or SRA."
+                  "All samples from this source database will have their metadata refreshed."))
+
+    def handle(self, *args, **options):
+        """Refreshes the metadata for RNA-Seq samples
```
Comment (kurtwheeler) [inline foreman/data_refinery_foreman/foreman/management/commands/update_experiment_metadata.py]: This docstring and the helpstring above both say samples, but this doesn't update the samples, it does the experiments. This docstring also specifies RNA-Seq but I think it'll work on all experiments.

Comment (wvauclain) [reply foreman/data_refinery_foreman/foreman/management/commands/update_experiment_metadata.py]: My bad, I originally copied the refresh samples file for this, and I must not have changed all of the comments

File: foreman/data_refinery_foreman/foreman/management/commands/test_update_experiment_metadata.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            Experiment.objects.get_or_create(accession_code=experiment.accession_code)[0].title,
+            BAD_TITLE
+        )
+
+        # Run the command again to make sure that it does not fail if there are no changes
+        command = Command()
+        command.handle()
+
+    def test_geo_experiment_missing_metadata(self):
+        """Tests that an SRA experiment has its missing metadata added."""
```
Comment (kurtwheeler) [inline foreman/data_refinery_foreman/foreman/management/commands/test_update_experiment_metadata.py]: This dockstring should say GEO

File: foreman/data_refinery_foreman/foreman/management/commands/test_update_experiment_metadata.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+            Experiment.objects.get_or_create(accession_code=experiment.accession_code)[0].title,
+            BAD_TITLE
+        )
+
+        # Run the command again to make sure that it does not fail if there are no changes
+        command = Command()
+        command.handle()
+
+    def test_array_express_experiment_missing_metadata(self):
+        """Tests that an SRA experiment has its missing metadata added."""
```
Comment (kurtwheeler) [inline foreman/data_refinery_foreman/foreman/management/commands/test_update_experiment_metadata.py]: This Docstring should say Array Express

File: foreman/data_refinery_foreman/surveyor/array_express.py [hunk 5]
```diff
@@ -42,54 +42,30 @@ class ArrayExpressSurveyor(ExternalSourceSurveyor):
...
-            raise UnsupportedPlatformException
         # If there is more than one arraydesign listed in the experiment
         # then there is no other way to determine which array was used
         # for which sample other than looking at the header of the CEL
         # file. That obviously cannot happen until the CEL file has been
         # downloaded so we can just mark it as UNKNOWN and let the
         # downloader inspect the downloaded file to determine the
         # array then.
-        elif len(parsed_json["arraydesign"]) != 1 or "accession" not in parsed_json["arraydesign"][0]:
+        if len(parsed_json["arraydesign"]) != 1\
```
Comment (kurtwheeler) [inline foreman/data_refinery_foreman/surveyor/array_express.py]: You seem to be removing the check that parsed_json actually contains a key called `arraydesign`. I don't know how consistent their json is, so I'd prefer to keep such safety checks in place.

Comment (kurtwheeler) [reply foreman/data_refinery_foreman/surveyor/array_express.py]: Oh wait I see, you moved it to another place :)

**Pull request merged**